code/_onclick/click.dm

```diff
@@ -32,9 +32,13 @@
 */
 /atom/Click(location,control,params)
 	usr.ClickOn(src, params)
+
 /atom/DblClick(location,control,params)
 	usr.DblClickOn(src,params)
 
+/atom/MouseWheel(delta_x,delta_y,location,control,params)
+	usr.MouseWheelOn(src, delta_y)

```

Comment:
Pass delta_x and params

Then you can do params2list(params) to get advanced details like if control was pressed. 

see here: https://github.com/tgstation/tgstation/blob/master/code/_onclick/click.dm#L60

(assuming you want to do control detection)